runtime/near-vm-runner/src/imports.rs

```diff
@@ -3,10 +3,18 @@ use near_vm_logic::VMLogic;
 
 use std::ffi::c_void;
 
-struct ImportReference(*mut c_void);
+#[derive(Clone)]
+pub struct ImportReference(pub *mut c_void);
```

Comment:
1. Is it safe to clone the pointer? Who is cleaning the memory?
2. Can we avoid `pub`? Would `pub(crate)` be enough?

Reply:
1. Yes. The pointer is convert from VMLogic object. clone is to clone VMLogic object into wasmer memory. Wasmer does the cleaning on wasm runner exit.
2. No. Because this macro:
```
error[E0446]: crate-visible type `NearWasmerEnv` in public interface
   --> runtime/near-vm-runner/src/imports.rs:59:17
    |
13  |   pub(crate) struct NearWasmerEnv {
    |   ---------- `NearWasmerEnv` declared as crate-visible
...
59  |                   pub fn $func(env: &NearWasmerEnv, $( $arg_name: $arg_type ),* ) -> VMResult<($( $returns ),*)> {
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-visible type
...
175 | / wrapped_imports! {
176 | |     // #############
177 | |     // # Registers #
178 | |     // #############
...   |
319 | |     // #["protocol_feature_evm", EVM] test_api<[a: u64] -> []>,
320 | | }
    | |_- in this macro invocation
```
require NearWasmer and its fields to be pub. And therefore `    pub logic: ImportReference,` cannot be pub(crate)` because of "   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak crate-visible type"

Reply:
This actually is implicitly "pub(crate)" because it's in `mod imports;` which is not pub, therefore other crate cannot import it.